### ISSUE TITLE
Standardize GuardrailOutput: concrete model, required verdict, canonical risk score, per-category results

### ISSUE THREAD

**Issue by dni138** [mozilla-ai/any-guardrail#177]
## Motivation

`GuardrailOutput[ValidT, ExplanationT, ScoreT]` is generic with three optional fields. Across 17 guardrails we found ~10 incompatible instantiations:

| Guardrail | valid | explanation | score |
|---|---|---|---|
| Deepset / InjecGuard / Jasper / Pangolin / Protectai / Sentinel | bool | None | max-prob of *predicted* label (direction flips with the prediction) |
| HarmGuard | bool | None | P(unsafe) ✓ |
| ShieldGemma | bool | None | P(violation) ✓ |
| DuoGuard | bool | dict[str, bool] (12 categories) | max sigmoid prob |
| OffTopic (jina/stsb) | bool | dict[str, float] (both probs) | **None despite float hint (bug)** |
| AzureContentSafety | bool | dict[str, int\|list\|None] (severities + blocklist) | severity 0–7 (max/avg) |
| GraniteGuardian | bool \| None | full generation | `"yes"`/`"no"` **string** |
| LlamaGuard | bool | raw generation (S-codes unparsed) | None |
| Glider | **always None** | full generation | rubric int |
| Flowjudge | **always None** | feedback | rubric int |
| AnyLlm | bool | str | 0–1 confidence (**schema declares int — bug**) |
| Alinia | bool | entire API JSON | dict[str, dict[str, float]] |

`result.score > 0.8` means six different things; judges never commit to a verdict; DuoGuard's 12 probabilities and Llama Guard's S-codes are discarded or buried. This defeats the library's core promise: swapping guardrails without changing application code.

Design references researched: Guardrails AI (`FailResult{error_message, fix_value, error_spans}`, `validation_passed`, OnFailAction as *caller* policy), NeMo Guardrails (PASSED/MODIFIED/BLOCKED, per-rail logs), mozilla-ai/encoderfile proto (`SequenceClassificationResult{logits, scores, predicted_index, predicted_label}`, `TokenClassification` spans, `model_id` envelope), OpenAI omni-moderation (13 categories @ 0–1), Azure Content Safety (severity 0–7), Llama Guard S1–S14 / MLCommons, Granite Guardian probability-of-risk, Perspective spanScores, WildGuard, llm-guard `(sanitized, is_valid, risk_score)`.

## New standard (v0.6.0)

Concrete, non-generic `GuardrailOutput`:

- `valid: bool` — REQUIRED verdict. Judges map rubric → verdict via a new `pass_threshold` constructor arg (+ `higher_is_better`). Unparseable judge output fails closed (`valid=False`, `extra={"parse_failure": True}`).
- `score: float | None` — canonical risk: ~[0,1], higher = more likely violating. Injection classifiers report P(injection); Azure reports severity/7; rubric ints are NOT risk scores → `score=None` for judges (rubric preserved in `extra`).
- `categories: list[CategoryResult]` — `{name, description, triggered, score, severity}`. DuoGuard's 12 probabilities, Azure's 4 severities, Llama Guard's parsed S1–S14, binary classifiers' full label distribution, OffTopic's both-class probs.
- `spans: list[SpanResult] | None` — future-proofing (PII/NER, encoderfile TokenClassification, Perspective spanScores).
- `modified_text: str | None` — future sanitizers; action is implicit (`is not None`).
- `usage: GuardrailUsage | None` — model_id, latency_ms, token counts; stamped centrally.
- `explanation: str | None` — human-readable rationale ONLY.
- `extra: dict | None` + `raw: Any | None` — escape hatches.

Supporting provider change: the uniform `infer()` dict gains `labels: list[str] | None` (full ordered label list; HF via `id2label`, encoderfile `None` with a 2-class complement fallback — exposing labels from the binary is a suggested upstream encoderfile enhancement).

## Migration (before → after)

Binary injection classifier:
```python
# 0.5.x: score = confidence of predicted label (ambiguous)
# 0.6.0: score = P(injection); full distribution in categories
risky = result.score > 0.5                      # now always means "risky"
```

Multi-category (DuoGuard / Azure):
```python
# 0.5.x: result.explanation["Hate"] / result.explanation["hate"]
flagged = [c.name for c in result.categories if c.triggered]   # 0.6.0
```

Judges (Glider / Flowjudge):
```python
# 0.5.x: result.valid is None; rubric in result.score
g = Glider(pass_criteria=..., rubric=..., pass_threshold=3)    # 0.6.0: required
result.valid                                                    # bool
result.extra["rubric_score"]
```

GraniteGuardian:
```python
# 0.5.x: result.score == "yes"
# 0.6.0:
result.valid is False; result.extra["raw_answer"] == "yes"
```

Alinia: full JSON moved `explanation` → `raw`.

## Alternatives considered

- **Keep generics**: preserves per-guardrail "precision" but is the root cause — callers can't write guardrail-agnostic code; 10 shapes is the status quo failure.
- **Tri-state verdict enum (PASS/FAIL/UNKNOWN)**: pushes ambiguity to every call site; fail-closed bool + `extra["parse_failure"]` keeps the common path trivial.
- **Parallel class + deprecation shim**: pre-1.0, 17 call sites, doubles test surface for one release.

## Out of scope

- Input-side `validate()` signature standardization (str vs messages vs eval dicts).
- Streaming validation.
- Probability scores for LLM judges via logprobs (Granite/ShieldGemma-style) — future enhancement.
- Encoderfile binary exposing the full label list (upstream; complement fallback used meanwhile).
